pyccel/ast/class_defs.py

```diff
@@ -138,18 +138,18 @@
 
 #=======================================================================================
 
-ListClass = ClassDef('list', class_type = NativeHomogeneousList(),
+ListClass = ClassDef('list',
         methods=[
             PyccelFunctionDef('append', func_class = ListAppend),
-            PyccelFunctionDef('clear', func_class = ListClear,),
+            PyccelFunctionDef('clear', func_class = ListClear),
             PyccelFunctionDef('extend', func_class = ListExtend),
             PyccelFunctionDef('insert', func_class = ListInsert),
             PyccelFunctionDef('pop', func_class = ListPop),
         ])
```

Comment:
Why is the argument `class_type` not passed to the constructor anymore?

Reply:
In the new implementation the class type contains all information about the type. So it is no longer possible to initialise `NativeHomogeneousList()`, instead we need to do something like `NativeHomogeneousList(PythonNativeInt())` or `NativeHomogeneousList(NumpyNDArray(PythonNativeFloat()))`. As this class is valid for all the different kinds of lists that you can imagine I thought it was better to not provide this argument rather than creating a duplicate of this class on the fly for each type that is instantiated.